concurrent/src/main/scala/scalaz/concurrent/Timer.scala

```diff
@@ -103,7 +103,7 @@ case class Timer(timeoutTickMs: Int = 100, workerName: String = "TimeoutContextW
     futureNondeterminism.choose(timeoutFuture, future).map(_.fold(_._1.left, _._2.right))
   }
 
-  def withTimeout[T](task: Task[T], timeout: Long): Task[Timeout \/ T] = {
+  def withTimeout[T](task: Task[T], timeout: Long, workaroundForDoubleDefinition: Unit = ()): Task[Timeout \/ T] = {
```

Comment:
What's the pros and cons of `DummyImplicit` in comparison with default parameter?